### PR TITLE
Bump version to 16.0.3 (C++17 support)

### PR DESCRIPTION
**Pull request by andreittr**
Updated lib-compiler-rt to the latest upstream version; based off earlier work from @mschlumpp & @marcrittinghaus .
Part of a patch set w/ further PRs under [libcxx](https://github.com/unikraft/lib-libcxx/pull/27), [libcxxabi](https://github.com/unikraft/lib-libcxxabi/pull/3), [libunwind](https://github.com/unikraft/lib-libunwind/pull/6) & [lib-musl](https://github.com/unikraft/lib-musl/pull/45).

### CODE REVIEW

File: Config.uk [hunk 1]
```diff
@@ -1,13 +1,9 @@
...
-    select LIBPOSIX_SYSINFO
-    select LIBUKMMAP
-    select LIBUNWIND
-    default n
+	bool "compiler-rt - Compiler runtime support"
+	select LIBPOSIX_SYSINFO
+	select LIBUNWIND
+	default n
 
 if LIBCOMPILER_RT
```
Comment (StefanJum) [inline Config.uk]: Is there a reason this `if` condition is still present? It doesn't seem to do anything.

**Review by StefanJum — CHANGES_REQUESTED** [review]
Thank you @andreittr. besides the few inline comments:

* Please change the commits to match the [`unikraft`  commit format](https://unikraft.org/docs/contributing/submitting-changes/#commits). (i.e. sign-off them and add the `co-authored-by` and `sign-off` of the co-authors, add the `component/selector` to the commit message).
* Why are there so many changes to the `Makefile.uk` file? It looks like the source files are just reordered, removed from one place and added again. Is this intended (is there a specific order it follows)?

Besides that, functionally it seems to work fine, I'll keep testing.

**Comment by jschlumpp** [comment]
> Why are there so many changes to the `Makefile.uk` file? It looks like the source files are just reordered, removed from one place and added again. Is this intended (is there a specific order it follows)?

I'm pretty sure that Marc sorted the file list lexicographically.

**Comment by StefanJum** [comment]
> I'm pretty sure that Marc sorted the file list lexicographically.

Yeah that makes sense, thanks.

**Comment by andreittr** [comment]
Building libcxx v16 requires GCC >= 12 and that is a bit too restrictive a change for now.
I'm going to close this set of PRs and resubmit new ones that bump LLVM only to 14, along with the minimal necessary changes to get it working. libcxx 14 _does_ compile with GCC 11 (though not with earlier versions :( ).

**Pull request closed without merging**